Pimp codeexamples (issue #640) part 2 (of pull request #711)

Mo-Gul opened the pull request:
basically this is the continuation of pull request #711, but it doesn't show up there, because I first merged branch `PimpCodeexamples` into `master`.

hmenke commented:
Does this replace #711 or is it in addition?

hmenke commented:
BTW, the commit 4dd4ac8cddf37cf553a9cbc92522d92d0ca75ab5 was the one that screwed everything up.

Mo-Gul commented:
Regarding the *code* this seems to be a *replacement*, but of course the *conversation* *continues* here, if you still have something ...

I had to create a new Pull Request, because this time I first merged `PimpCodeexamples` into (my) `master` and thus the newest commit didn't show up in pull request #711. (This is in line with the text shown near the end at that pull request

> Add more commits by pushing to the [**PimpCodeexamples**](https://github.com/Mo-Gul/pgf/tree/PimpCodeexamples) branch on [**Mo-Gul/pgf**](https://github.com/Mo-Gul/pgf).

hmenke commented:
I will probably merge this PR step-by-step because I really want to avoid merging 4dd4ac8.  That will just screw up the history again, like this did https://github.com/pgf-tikz/pgf/compare/fcba572...9d5755f (which caused #694).

Mo-Gul commented:
As you wish. That is why I asked you (several times) to do the merge in my fork, but after I didn't hear anything from you, I gave it a try myself ...

As also already written via email, I have added you as a collaborator in my fork, so if you can fix the mess there ... You decide what is less work for you.

hmenke commented:
I have rebase your `master` onto the upstream `master` and removed the broken stuff, so it will never show up in the git history.  I've opened #733 for that.  Could you please review it, to make sure that I haven't accidentally dropped any of your changes?

Mo-Gul committed:
- corrected a word in `pgfmanual-en-dv-axes.tex`
- harmonized spelling of `i.e.` and `e.g.`
- corrected line breaking in `pgfmanual-en-math-numberprinting.tex` where a |...| was split across lines

Mo-Gul committed:
- checked `pattern.meta` library stuff and fixed some minor issues

hmenke commented:
As I wrote previously, I have rebased this branch onto master, so it is not longer mergable.  Please reset this branch to `origin/master` and continue from there.  To do so, follow these steps:
```bash
# Add the upstream source as a remote repository to git
git remote add upstream https://github.com/pgf-tikz/pgf.git
# Fetch the latest pgf-tikz/master branch
git fetch upstream master
# Rename your current master branch into oldmaster
git branch -m master oldmaster
# Check out the pgf-tikz/master branch under the name master
git checkout -B master upstream/master
# Cherry pick the two new commits you made on oldmaster to the current master
git cherry-pick 329e374 1a5d61d
# Force push to the remote because the refs have changed
git push --force origin master
```

Mo-Gul commented:
Done. Many thanks for the instructions.

Please keep in mind that I am not a professional in git, thus, e.g. I don't have a clue what "rebasing" or a "cherry-pick" is. But of course I am willing to learn, so keep going with instructions ;)

hmenke commented on doc/generic/pgf/text-en/pgfmanual-en-library-patterns.tex:
```diff
@@ -91,80 +92,81 @@ \subsection{User-Defined Patterns}
-\begin{key}{/pgf/patterns/name=\meta{name}}
-    The name of the pattern by which it can be used later on.
-\end{key}
+    keys and values to define the pattern. The following keys are available:
```
Stop changing the indentation!

Mo-Gul replied:
Nope. That is one of the only things I can do.

We keep repeating ourselves. Bullshit happened when I did that to "thousand" files. But when only checking this very small pieces it is very very unlikely that bullshit will happen again. While reformatting I am *reading* the text, too, so it is kind of double-checked. And before committing it is checked again that no bullshit happened ...

Mo-Gul commented:
Can't this pull request simply be closed? Because all new commits are not related to issue #640 any more ...

hmenke commented:
It's up to you.  You could also just change the title and I delete the obsolete comments.

Mo-Gul commented:
Then we'll close it. Changing the title may cause some confusion because this issue is referenced elsewhere (I guess).

The real pull request with these changes was done by Henry in pull request #733.
